identity/Cargo.toml

```diff
@@ -1,6 +1,6 @@
 [package]
 name = "libp2p-identity"
-version = "0.2.2"
+version = "0.2.3"
```

Comment:
Needs a bump of `libp2p-identify` in the root `Cargo.toml`. See CI failure: https://github.com/libp2p/rust-libp2p/actions/runs/5891817364/job/15979771326?pr=4337

Reply:
Maybe cargo-semver-check does not adhere to the override?

https://github.com/libp2p/rust-libp2p/blob/cbdbaa836e1159fb4e8b20e76c32f1af5ec66926/Cargo.toml#L111-L118

Reply:
I am surprised that this is showing up now, how did we ever bump this version? Maybe this failure happens now with the recent version change of `cargo semver-checks`?

The issue with `libp2p-identify` is that we need to patch it across the workspace and I'd assume that the temporary workspace created by `cargo semver-checks` does not support that. To properly fix this, we should move `libp2p-identity` out of the mono-repo. We should very rarely make breaking changes to it anyway because it sets at the very bottom of the dependency chain of the ecosystem.

Reply:
> The issue with `libp2p-identity` is that we need to patch it across the workspace and I'd assume that the temporary workspace created by `cargo semver-checks` does not support that.

Maybe @obi1kenobi you have seen this issue before, i.e. that `cargo-semver-checks` does not adhere to the workspace `patch`es. Note that this is only a suspicion. Issue might as well be on the rust-libp2p end.

---

@thomaseizinger @jxs I am fine merging here as is, i.e. with the failing `cargo-semver-checks` check on the `libp2p-identity` run. Once `libp2p-identity` `v0.2.3` is released we can update our root `Cargo.toml`. Objections?

Reply:
Yeah I don't believe it copies patches. You can double-check and make sure by looking up the generated manifest inside `target/semver-checks`. But I'm pretty sure we never read patches from the source manifest so that's probably it.

If running `cargo-semver-checks` on an intermediate state that isn't right before a `cargo publish`, checking with patches included makes sense.

But is there risk that pre-publish checking while using patches might falsely say there are no semver issues, only for issues to come up after the crate gets published and used by users that don't have the patches locally?

Reply:
updated main `Cargo.toml` to `identity` `0.2.3`